app/models/group_order_invoice.rb

```diff
@@ -0,0 +1,60 @@
+class GroupOrderInvoice < ApplicationRecord
+
+  belongs_to :group_order
+  validates_presence_of :group_order
+  validates_uniqueness_of :invoice_number
+  validate :tax_number_set
+  after_initialize :init, unless: :persisted?
+
+  def generate_invoice_number(count)
+    trailing_number = count.to_s.rjust(4, '0')
+    unless  GroupOrderInvoice.find_by(invoice_number: Time.now.strftime("%Y%m%d") + trailing_number)
+      Time.now.strftime("%Y%m%d") + trailing_number
+    else
+      generate_invoice_number(count.to_i + 1)
+    end
+  end
+
+  def tax_number_set
+    if FoodsoftConfig[:contact][:tax_number].blank?
+      errors.add(:group_order_invoice, "Keine Steuernummer in FoodsoftConfig :contact gesetzt")
+    end
```

Comment:
this method does not access the database and therefore should be removed from the model and move to a better location (BTW: there are user visible string which are not translated)